Feature/ot111 18

denisdimarco opened the pull request:
[https://alkemy-labs.atlassian.net/browse/OT111-18](https://alkemy-labs.atlassian.net/browse/OT111-18)

- Created Category Entity
- Created Category Script
- Created Category Repository

- Added Gradle dependencies and updated application.yml.

denisdimarco committed:
Added dependencies and updated application.yml

JPA, HibernateValidator, SpringWeb, Postgresql.

denisdimarco committed:
Create Category.sql

denisdimarco committed:
Create CategoryEntity

denisdimarco committed:
Create CategoryRepository

OsvaldoGWP commented on src/main/java/com/alkemy/ong/model/entity/CategoryEntity.java:
```diff
@@ -0,0 +1,44 @@
+
+    private String modifiedBy;
+
+    @NotNull
```
este notnull no aplica para tipos de datos primitivos

denisdimarco replied:
- [x] @NotNull deleted

OsvaldoGWP commented on src/main/java/com/alkemy/ong/model/entity/CategoryEntity.java:
```diff
@@ -0,0 +1,44 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
```
para los string podemos usar @NotEmpty ya q aparte de controllar q no sea nulo tambien controla q no sea puro espacios en blancos

denisdimarco replied:
- [x] Replaced `@NotNull -->  @NotEmpty` in String fields

OsvaldoGWP commented on src/main/java/com/alkemy/ong/model/entity/CategoryEntity.java:
```diff
@@ -0,0 +1,44 @@
+    private String image;
+
+    @NotNull
+    private OffsetDateTime createdDate;
```
este campo lo inicialicemos por defecto

denisdimarco replied:
- [x] Removed @NotNull in `private OffsetDateTime createdDate;`

OsvaldoGWP commented on build.gradle:
```diff
@@ -20,9 +20,18 @@ repositories {
 
 dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter'
```
IMPORTANTE: actualizar con develop porq ya va haber conflictos cuando se empiece a aprobar los otros pr y a mergear

OsvaldoGWP commented on build.gradle:
```diff
@@ -20,9 +20,18 @@ repositories {
+	runtimeOnly 'org.postgresql:postgresql'
 	annotationProcessor 'org.projectlombok:lombok'
 	testImplementation 'org.springframework.boot:spring-boot-starter-test'
+	testImplementation 'org.springframework.boot:spring-boot-starter-test'
```
dependencia repetida

denisdimarco replied:
Removed duplicated dependency.

denisdimarco committed:
Merge branch 'develop' into feature/OT111-18

denisdimarco committed:
OT111-18 - Fixed PR

Replaced @NotNull by @NotEmpty
Removed @NotNulls 
Removed Duplicated Dependencies

denisdimarco committed:
Merge branch 'feature/OT111-18' of https://github.com/alkemyTech/OT111-Server into feature/OT111-18

denisdimarco committed:
Merge branch 'develop' into feature/OT111-18

denisdimarco committed:
OT111-18 -  Change Annotations in MemberEntity 

Fix annotation of OT111-22
